velox/exec/Unnest.cpp

```diff
@@ -98,11 +99,17 @@ void Unnest::addInput(RowVectorPtr input) {
     auto currentSizes = rawSizes_[channel];
     auto currentIndices = rawIndices_[channel];
     for (auto row = 0; row < size; ++row) {
-      if (!currentDecoded.isNullAt(row)) {
-        auto unnestSize = currentSizes[currentIndices[row]];
-        if (rawMaxSizes_[row] < unnestSize) {
-          rawMaxSizes_[row] = unnestSize;
-        }
+      vector_size_t unnestSize = -1;
+      if (outer_ &&
+          (currentDecoded.isNullAt(row) ||
+           currentSizes[currentIndices[row]] == 0)) {
+        unnestSize = 1;
+      } else if (!currentDecoded.isNullAt(row)) {
+        unnestSize = currentSizes[currentIndices[row]];
+      }
+
+      if (rawMaxSizes_[row] < unnestSize) {
+        rawMaxSizes_[row] = unnestSize;
```

Comment:
When outer_ is false, and `currentDecoded.isNullAt(row)`, unnestSize is -1. Would there be any problem?

Reply:
It's safe since `rawMaxSizes_[row]` is 0 by default. The test `arrayWithOrdinality` also guarantee this situation.